### PR TITLE
Seperating out linerperforation for well and activity submission

### PR DESCRIPTION
**Pull request by Sybrand**
- Removed nose (nose is no longer actively maintained, we're not using it's output, and it was throwing exceptions while handling test cases).
- Migrate existing liner perforations, to activity submission liner perforation where valid.
- Added various tests.

### CODE REVIEW

**Comment by Sybrand** [comment]
[EPIC#Submission | Stop creating duplicate records in the liner_perforation table |3| !!!](https://trello.com/c/VG8swyKR/1815-epicsubmission-stop-creating-duplicate-records-in-the-linerperforation-table-3)

**Comment by Sybrand** [comment]
[EPIC#Submission | Stop creating duplicate records in the liner_perforation table |3| !!!](https://trello.com/c/VG8swyKR/1815-epicsubmission-stop-creating-duplicate-records-in-the-linerperforation-table-3)

File: app/backend/wells/migrations/0083_merge_20190510_2215.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Generated by Django 2.2.1 on 2019-05-10 22:15
```
Comment (stephenhillier) [inline app/backend/wells/migrations/0083_merge_20190510_2215.py]: I think this file is redundant and will need another merge migration if it's added to release.  Can you try merging in release into this branch?

File: app/backend/wells/migrations/0084_move_liner_perforations.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+from django.db import migrations
+
+
+def move_liner_perforations(apps, schema_editor):
+    LinerPerforation = apps.get_model('wells', 'LinerPerforation')
+    ActivitySubmissionLinerPerforation = apps.get_model('wells', 'ActivitySubmissionLinerPerforation')
+    ActivitySubmission = apps.get_model('wells', 'ActivitySubmission')
+
+    for perforation in LinerPerforation.objects.filter(activity_submission__isnull=False):
+        # Create a new perforatin object.
```
Comment (stephenhillier) [inline app/backend/wells/migrations/0084_move_liner_perforations.py]:
```suggestion
        # Create a new perforation object.
```

File: app/backend/wells/migrations/0083_merge_20190510_2215.py (hunk 1 above)
Comment (Sybrand) [reply app/backend/wells/migrations/0083_merge_20190510_2215.py]: Afraid it's not easy - there's a whole string of dependancies to it now.

**Pull request merged**